### PR TITLE
Custom message in email

### PR DESCRIPTION
**Pull request by Facyla**
Here is a PR corresponding to https://github.com/arckinteractive/upload_users/issues/16

This was implemented as a plugin setting, which enables to define a custom message that will be inserted after the main email notification content. Plugin settings also provide a preview of the full message body.
Admins can welcome people, recommend groups, invite to read a specific resource, etc.

### CODE REVIEW

File: classes/UploadUsers.php [hunk 1]
```diff
@@ -318,7 +318,8 @@ function uploadUser($record) {
 
 				if ($this->notification) {
 					$subject = elgg_echo('upload_users:email:subject', array(elgg_get_config('sitename')));
-					$message = elgg_echo('upload_users:email:message', array($record['name'], elgg_get_config('sitename'), $record['username'], $record['password'], elgg_get_site_url()));
+					$custom_message = elgg_get_plugin_setting('custom_message', 'upload_users');
```
Comment (hypeJunction) [inline classes/UploadUsers.php]: Default this to an empty string to avoid problems with `sprintf`

File: languages/en.php [hunk 2]
```diff
@@ -55,6 +55,8 @@
 	'upload_users:setting:update_existing_users' => 'Update profile information when user account already exists',
 	'upload_users:setting:fix_usernames' => 'Fix usernames if the value does not meet Elgg standards (e.g. contains special characters), and suffix them with a number if already taken',
 	'upload_users:setting:fix_passwords' => 'Generate a new cleartext passwords, if the value does not meet Elgg standards (e.g. is too short)',
+	'upload_users:notification:custom' => "Custom message inserted withing the email notification",
```
Comment (hypeJunction) [inline languages/en.php]: within

File: views/default/plugins/upload_users/settings.php [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<?php
+
+// Custom message in email notification
+echo '<p><label>' . elgg_echo('upload_users:notification:custom'). '</label> ' . elgg_view('input/longtext', array('name' => 'params[custom_message]', 'value' => $vars['entity']->custom_message)) . '</p>';
```
Comment (hypeJunction) [inline views/default/plugins/upload_users/settings.php]: One tag per line. It's more readable if you use inline html and then echo php strings:

```php
?>
<div>
   <label><?php echo 'label' ?></label>
   <?php echo $input ?>
</div>
```

**Comment by Facyla** [comment]
Read comments and updated code structure
Set up new default settings https://github.com/arckinteractive/upload_users/issues/14
And also add some spacing between label and select input

Also there are now 2 missing Finnish strings : 
	'upload_users:notification:custom' => "Custom message inserted within the email notification",
	'upload_users:notification:preview' => "Preview notification message",

File: classes/UploadUsers.php [hunk 4]
```diff
@@ -318,7 +318,9 @@ function uploadUser($record) {
...
+					if (!$custom_message) { $custom_message = ''; }
```
Comment (hypeJunction) [inline classes/UploadUsers.php]: Tertiary operations should span across multiple lines.
In this particular case, `elgg_get_plugin_setting()` takes a third argument for `$default` value, so you don't need to use `if`

File: views/default/forms/upload_users/upload.php [hunk 5]
```diff
@@ -14,13 +14,13 @@
  */
 // CSV File Input
 echo '<div>';
-echo '<label>' . elgg_echo('upload_users:choose_file') . '</label>';
+echo '<label>' . elgg_echo('upload_users:choose_file') . '</label> ';
```
Comment (hypeJunction) [inline views/default/forms/upload_users/upload.php]: Please no spaces after the tag. Use CSS if you need to pad.

File: views/default/plugins/upload_users/settings.php [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+// Custom message in email notification
+
+// Preview notification message
+$preview_message = elgg_echo('upload_users:email:message', array("[NAME]", elgg_get_config('sitename'), "[USERNAME]", "[PASSWORD]", elgg_get_site_url(), $vars['entity']->custom_message));
+?>
+
+<div>
+	<p>
+		<label><?php echo elgg_echo('upload_users:notification:custom'); ?></label>
+		<?php echo elgg_view('input/longtext', array('name' => 'params[custom_message]', 'value' => $vars['entity']->custom_message)); ?>
```
Comment (hypeJunction) [inline views/default/plugins/upload_users/settings.php]: Don't be afraid to use line breaks ;)

```php
<?php
echo elgg_view('input/longtext', array(
   'name' => 'params[custom_message]', 
   'value' => $vars['entity']->custom_message
)); 
?>
```

**Comment by hypeJunction** [comment]
Looking good. If you can fix the coding style, I can merge.

**Comment by Facyla** [comment]
Thanks for the comments, here is the updated version !

I know i'm not much a line-break person, but i'm working on that^^

**Comment by hypeJunction** [comment]
Thanks.

**Comment by Facyla** [comment]
You're welcome :)

**Pull request merged**